stress/common/src/main/java/alluxio/stress/client/ClientIOTaskResult.java

```diff
@@ -0,0 +1,351 @@
+    return errors;
+  }
+
+  @Override
+  public TaskResult.Aggregator aggregator() {
+    return new Aggregator();
+  }
+
+  private static final class Aggregator implements TaskResult.Aggregator {
+    @Override
+    public ClientIOTaskResult aggregate(Iterable<TaskResult> results) throws Exception {
+      Iterator<TaskResult> it = results.iterator();
+      if (it.hasNext()) {
+        TaskResult taskResult = it.next();
+        if (it.hasNext()) {
+          throw new IOException(
+              "ClientIO is a single node test, so multiple task results cannot be aggregated.");
+        }
+        if (!(taskResult instanceof ClientIOTaskResult)) {
+          throw new IOException(
+              "TaskResult is not of type ClientIOTaskResult. class: " + taskResult.getClass()
+                  .getName());
+        }
+        return (ClientIOTaskResult) taskResult;
+      }
+      return new ClientIOTaskResult();
+    }
+  }
+
+  @Override
+  public alluxio.stress.GraphGenerator graphGenerator() {
+    return new GraphGenerator();
+  }
+
+  /**
+   * The graph generator for this summary.
+   */
+  public static final class GraphGenerator extends alluxio.stress.GraphGenerator {
+    @Override
+    public List<Graph> generate(List<? extends Summary> results) {
+      List<Graph> graphs = new ArrayList<>();
+      // only examine ClientIOSummary
+      List<ClientIOTaskResult> summaries =
+          results.stream().map(x -> (ClientIOTaskResult) x).collect(Collectors.toList());
+
+      // Iterate over all operations
+      for (ClientIOOperation operation : ClientIOOperation.values()) {
+        for (Boolean readRandom : Arrays.asList(false, true)) {
+          List<ClientIOTaskResult> opSummaries =
+              summaries.stream().filter(x -> x.mParameters.mOperation == operation)
+                  .filter(x -> x.mParameters.mReadRandom == readRandom)
+                  .collect(Collectors.toList());
+
+          if (!opSummaries.isEmpty()) {
+            // first() is the list of common field names, second() is the list of unique field names
+            Pair<List<String>, List<String>> fieldNames = Parameters.partitionFieldNames(
+                opSummaries.stream().map(x -> x.mParameters).collect(Collectors.toList()));
+
+            // Split up common description into 100 character chunks, for the sub title
+            List<String> subTitle = new ArrayList<>(Splitter.fixedLength(100).splitToList(
+                opSummaries.get(0).mParameters.getDescription(fieldNames.getFirst())));
+
+            for (ClientIOTaskResult summary : opSummaries) {
+              String series = summary.mParameters.getDescription(fieldNames.getSecond());
+              subTitle.add(series + ": " + DateFormat.getDateTimeInstance()
+                  .format(summary.computeLastEndMs()));
+            }
+
+            LineGraph responseTimeGraph = new LineGraph(
+                operation + (readRandom ? " - Random" : " - Sequential") + " - Throughput",
```

Comment:
hmmmm this `Random`/`Sequential` thing again. It sounds like we need to come up with some kind of class/primitive that can take a parameter name and provide some interface to give better human-readable values.

Reply:
I updated this